File: backend/examples/openrouter_api_opik_integration.ts
/*
 * File Name: openrouter_api_opik_integration.ts
 * Description: This file contains the code for calling the openrouter api
 * Author Name: The Open Router Team
 * Creation Date: 19-Jan-2024
 * Modified Date: 24-Jan-2026
 * Version: 1.0
 *
 * Instructions to run:
 * Follow the command sequence below in the terminal:
 * 1. Install the required dependencies:
 * npm install @types/node
 * 2. Install typescript
 * npm install typescript
 * 3. Initialize typescript
 * npx tsc --init
 * 4. Compile the script:
 * tsc openrouter_api_opik_integration.ts
 * 5. Run the compiled JavaScript file:
 * node openrouter_api_opik_integration.js
 *
 * File Execution State: Not as intended
 * 
 * Note: Make sure you update the API key with yours. If you don't have one, then
 * please feel to create one from the website below.
 * https://openrouter.ai/settings/keys
 */

import OpenAI from "openai";
import { trackOpenAI } from "opik-openai";

(async() => {

    try
    {
        // Initialize the original OpenAI client
        const open_ai = new OpenAI({
            apiKey: '<API_KEY>'
        });

        // Wrap the client with opik tracking
        const trackedOpenAI = trackOpenAI(open_ai);

        // Use the tracked client just like the original
        const completion = await trackedOpenAI.chat.completions.create({
            model: "xiaomi/mimo-v2-flash:free",
            messages: [
                {
                    role: "user",
                    content: "Hello, how can you help me today?"
                }
            ]
        });

        console.log(completion.choices[0].message.content);

        // Ensure all traces are sent before your app terminates
        await trackedOpenAI.flush();
    }
    catch(error)
    {
        console.error('Error:', error);
    }
})();

File: backend/scripts/examples/openrouter_api_opik_integration.py
'''
File Name: openrouter_api_opik_integration.py
Description: This file contains the code for calling the openrouter api
Author Name: Open Router Team
Modified Date: 21-Jan-2026
Version: 1.0

Instructions to run:
Follow the command sequence below in the terminal:
1. Setup the virtual environment:
python3 -m venv <path/to/venv>
2. Activate the virtual environment:
source <path/to/venv>/bin/activate
3. Install required packages:
pip<3/x> install requests
4. Run the script:
python<version> openrouter_api_opik_integration.py
example: python3 openrouter_api_opik_integration.py

File Execution State: Execution successful but with model response is incomplete

Note: Make sure you update the API key with yours. If you don't have one, then
please feel to create one from the website below.
https://openrouter.ai/settings/keys
'''

from opik.integrations.openai import track_openai
from openai import OpenAI

# Initialize the OpenAI client with OpenRouter base URL
client = OpenAI(
    base_url="https://openrouter.ai/api/v1",
    api_key="<API_KEY>"
)
client = track_openai(client)

# Optional headers for OpenRouter leaderboard
headers = {
    "HTTP-Referer": "https://openrouter.ai/",  # Optional. Site URL for rankings
    "X-Title": "Open_Router"  # Optional. Site title for rankings
}

response = client.chat.completions.create(
    model="xiaomi/mimo-v2-flash:free",  # Specify the Xiaomi MIMO model
    extra_headers=headers,
    messages=[
        {"role": "user", "content": "You are a financial advisor and can you please suggest me with available investment options in the UK region?"}
    ],
    temperature=0.7,
    max_tokens=100
)

print(response.choices[0].message.content)
print('\n')
print('\n')
print('\n')
print(response)